ksql-common/src/main/java/io/confluent/ksql/util/SchemaUtil.java

```diff
@@ -298,25 +299,31 @@ public static String buildAvroSchema(final Schema schema, String name) {
   private static org.apache.avro.Schema getAvroSchemaForField(Schema fieldSchema) {
     switch (fieldSchema.type()) {
       case STRING:
-        return create(org.apache.avro.Schema.Type.STRING);
+        return getSchemaWithNullableCheck(create(org.apache.avro.Schema.Type.STRING));
       case BOOLEAN:
-        return create(org.apache.avro.Schema.Type.BOOLEAN);
+        return getSchemaWithNullableCheck(create(org.apache.avro.Schema.Type.BOOLEAN));
       case INT32:
-        return create(org.apache.avro.Schema.Type.INT);
+        return getSchemaWithNullableCheck(create(org.apache.avro.Schema.Type.INT));
       case INT64:
-        return create(org.apache.avro.Schema.Type.LONG);
+        return getSchemaWithNullableCheck(create(org.apache.avro.Schema.Type.LONG));
       case FLOAT64:
-        return create(org.apache.avro.Schema.Type.DOUBLE);
+        return getSchemaWithNullableCheck(create(org.apache.avro.Schema.Type.DOUBLE));
       default:
         if (fieldSchema.type() == Schema.Type.ARRAY) {
-          return createArray(getAvroSchemaForField(fieldSchema.valueSchema()));
+          return getSchemaWithNullableCheck(
+              createArray(getAvroSchemaForField(fieldSchema.valueSchema())));
         } else if (fieldSchema.type() == Schema.Type.MAP) {
-          return createMap(getAvroSchemaForField(fieldSchema.valueSchema()));
+          return getSchemaWithNullableCheck(
+              createMap(getAvroSchemaForField(fieldSchema.valueSchema())));
         }
         throw new KsqlException("Unsupported AVRO type: " + fieldSchema.type().name());
     }
   }
 
+  private static org.apache.avro.Schema getSchemaWithNullableCheck(org.apache.avro.Schema schema) {
```

Comment:
Now that we always allow nulls, a more appropriate name would be `unionWithNull` IMO.